tutorials/katacoda/thanos/3-receiver/step2.md

```diff
@@ -0,0 +1,98 @@
+</details>  
+
+## Thanos Receive
+
+Enter [Thanos Receive](https://thanos.io/tip/components/receive.md/).
+
+`Thanos Receive` is a component that implements the [Prometheus Remote Write API](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write). This means that it will accept metrics data that is sent to it by other Prometheus instances. 
+
+Prometheus can be configured to `Remote Write`. This means that Prometheus will send all of its metrics data to a remote endpoint as they are being ingested - useful for our requirements!
+
+In its simplest form, when `Thanos Receive` receives data from Prometheus, it stores it locally and exposes a `Store API` server so this data can be queried by `Thanos Query`.
+
+`Thanos Receive` has more features that will be covered in future tutorials, but let's keep things simple for now.
+
+## Run Thanos Receive
+
+The first component we will run in our new architecture is `Thanos Receive`:
+
+```
+docker run -d --rm \
+    -v $(pwd)/receive-data:/receive/data \
+    --net=host \
+    --name receive \
+    quay.io/thanos/thanos:v0.21.0 \
+    receive \
+    --tsdb.path "/receive/data" \
+    --grpc-address 0.0.0.0:10907 \
+    --http-address 0.0.0.0:10909 \
+    --label "receive_replica=\"0\"" \
+    --label "receive_cluster=\"wayne-enterprises\"" \
+    --remote-write.address 0.0.0.0:10908
+```{{execute}}
+
+Let's talk about some important parameters:
+* `--label` - `Thanos Receive` requires at least one label to be set. These are called 'external labels' and are used to uniquely identify this instance of `Thanos Receive`.
+* `--remote-write.address` - This is the address that `Thanos Receive` is listening on for Prometheus' remote write requests.
+
+Let's verify that this is running correctly. Since `Thanos Receive` does not expose a UI, we can check it is up by retrieving its metrics page.
+
+```
+curl http://127.0.0.1:10909/metrics
+```{{execute}}
+
+## Run Thanos Query
+
+Next, let us run a `Thanos Query` instance:
+
+```
+docker run -d --rm \
+    --net=host \
+    --name query \
+    quay.io/thanos/thanos:v0.21.0 \
+    query \
+    --http-address "0.0.0.0:39090" \
+    --store "0.0.0.0:10907"
+```{{execute}}
+
+`Thanos Receive` exposed its GRPC endpoint at `0.0.0.0:10907`, so we need to tell `Thanos Query` to use this endpoint with the `--store` flag.
+
+Verify that `Thanos Query` is working and configured correctly by looking at the 'stores' tab [here](https://[[HOST_SUBDOMAIN]]-39090-[[KATACODA_HOST]].environments.katacoda.com/stores).
+
+Now we are done right? Try querying for some data...
+
+![alt text](./assets/receive-empty-query-result.png)
+
+<details>
+ <summary>Uh-oh! Why are we seeing 'Empty Query Result' responses?</summary>
+
+We have correctly configured `Thanos Receive` & `Thanos Store`, but we have not yet configured Prometheus to write to remote write its data to the right place.
```

Comment:
```suggestion
We have correctly configured `Thanos Receive` & `Thanos Query`, but we have not yet configured Prometheus to write to remote write its data to the right place.
```

We are not using Thanos Store anywhere in this tutorial, did you meant Thanos Query?

Reply:
doh! yes of course